programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -203,6 +196,10 @@ static int run_opts(int argc, char *argv[]) {
     return optind;
 }
 
+static int dns_fallback(const char *name, void *ctx, struct in_addr* addr) {
```

Comment:
Are you thinking we will implement different versions of the fallback fn in ifdefs?

It feels like we need another layer of extensions for system/resolver-specific handling of apply() and fallback() at least.

Reply:
Why? What would another layer of extension do?  Can’t you do whatever you
need in your implementation of the callback?

Reply:
Yes. I would just prefer this callback (and perhaps other resolver-specific behaviors) to be implemented in the tsdk rather than in the tunneler app. For example it would be nice if ziti-edge-tunnel and ziti desktop edge (Mac) could use the same implementations.